scheduling/allocator.rb

```diff
@@ -315,56 +375,106 @@ def update(vm, vm_host)
   # The VmHostSliceAllocation is used when system is configured
   # to allocate VMs inside VmHostSlice. It wraps around the VmHostAllocation class
   # used otherwise.
-  # When creating a dedicated slice for a VM, as in the Standard family, we look
-  # for a place for a new slice and select a cpu set that will be used for that.
+  # This class handles two distinct cases - one when a dedicated slice
+  # is created for a VM (as in Standard family). In that case it always looks
+  # for a place for a new slice and finds a cpu set that will be used for that.
+  # The other case is when a shared slice can be used, meaning multiple VMs can be
+  # placed inside a slice. In that case the Allocation first looks if the candidate host
+  # already has a slice allocated that matches the family of the VM and that
+  # has enough room. If so, it will use that. Otherwise, a new slice will be
+  # created same way as in the 'dedicated' case.
+  # All this logic is split between 'select_existing_slice' and 'update' methods. The first
+  # one checks if the candidate_host can host a slice or has one already.
+  # The second one creates a slice if needed, once the host candidate is selected.
   class VmHostSliceAllocation
     def initialize(candidate_host, request, vm_host_allocations)
       @candidate_host = candidate_host
       @request = request
       @vm_host_allocations = vm_host_allocations
+
+      @existing_slice = if @request.require_shared_slice then select_existing_slice end
```

Comment:
```suggestion
      @existing_slice = select_existing_slice if @request.require_shared_slice
```

This is the preferred style for a single line `if`.  It's not technically the same, because it doesn't set `@existing_slice` in the `else` case, but that difference shouldn't affect this code.